src/matching.rs

```diff
@@ -537,19 +537,18 @@ fn matches_compound_selector_internal<E>(selector: &CompoundSelector<E::Impl>,
 /// will almost certainly break as elements will start mistakenly sharing styles. (See
 /// `can_share_style_with` in `servo/components/style/matching.rs`.)
 #[inline]
-pub fn matches_simple_selector<E>(selector: &SimpleSelector<E::Impl>,
-                                  element: &E,
-                                  shareable: &mut bool)
-                                  -> bool
-                                  where E: Element {
+pub fn matches_simple_selector<'a, E>(selector: &SimpleSelector<E::Impl>,
+                                      element: &'a E,
+                                      shareable: &mut bool)
+                                      -> bool
+                                     where E: Element {
```

Comment:
nit: Indentation (disregard this if it's fixed in next commits, I'm reviewing one by one).